### PR TITLE
unchecked warnings fixed

### PR DESCRIPTION
**Pull request by zishanbilal**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/794)

### CODE REVIEW

File: src/main/java/beam/agentsim/events/PathTraversalEvent.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public PathTraversalEvent(double time, Id<Vehicle> vehicleId, String vehicleType
         this.startY = startY;
         this.endX = endX;
         this.endY = endY;
-        this.attributes = new AtomicReference<Map<String, String>>(Collections.EMPTY_MAP);
+        this.attributes = new AtomicReference<>(Collections.EMPTY_MAP);
```
Comment (carloscaldas) [inline src/main/java/beam/agentsim/events/PathTraversalEvent.java]: Replace the constant by the method call `Collections.emptyMap()`.

File: src/main/java/beam/analysis/PathTraversalSpatialTemporalTableGenerator.java [hunk 2]
```diff
@@ -333,13 +329,13 @@ public void printDataToFile(String path) {
...
-                        sb.append(energyConsumption[i].get(r5Link.linkId, vehicleAndFuelType));
+                        sb.append(energyConsumption.get(i).get(r5Link.linkId, vehicleAndFuelType));
                         sb.append("\t");
                         sb.append(fuelType);
                         sb.append("\t");
-                        sb.append(Math.round(numberOfVehicles[i].get(r5Link.linkId, vehicleAndFuelType)));
+                        sb.append(Math.round(numberOfVehicles.get(i).get(r5Link.linkId, vehicleAndFuelType)));
                         sb.append("\t");
-                        sb.append(Math.round(numberOfPassengers[i].get(r5Link.linkId, vehicleAndFuelType)));
+                        sb.append(Math.round(numberOfPassengers.get(i).get(r5Link.linkId, vehicleAndFuelType)));
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/PathTraversalSpatialTemporalTableGenerator.java]: Two changes:
1-Extract the tab character as a local final variable or a class level constant
2-Consider chaining all appends without repeating `sb`

File: src/main/java/beam/analysis/PathTraversalSpatialTemporalTableGenerator.java [hunk 3]
```diff
@@ -420,7 +416,7 @@ public double getFuelUsageBasedOnStartEndCoordinates(double fuelEconomy, Map<Str
 
         double deltaX = (endCoord.getX() - startCoord.getX()) / (numberOfInterpolationLinksUsedForNonRoadModes - 1);
         double deltaY = (endCoord.getY() - startCoord.getY()) / (numberOfInterpolationLinksUsedForNonRoadModes - 1);
-        double energyConsumptionPerLink = energyConsumption[currentBinIndex].get(r5TransitLink.linkId, vehicleAndFuelType) / numberOfInterpolationLinksUsedForNonRoadModes;
+        double energyConsumptionPerLink = energyConsumption.get(currentBinIndex).get(r5TransitLink.linkId, vehicleAndFuelType) / numberOfInterpolationLinksUsedForNonRoadModes;
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/PathTraversalSpatialTemporalTableGenerator.java]: This line is too long. Consider breaking it

File: src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java [hunk 4]
```diff
@@ -253,14 +252,14 @@ JFreeChart getGraphic(String mode, int iteration) {
...
-    public void writeGraphic(Integer iteration, String mode) {
+    private void writeGraphic(Integer iteration, String mode) {
         try {
 
             String filename = fileName + "_" + mode + ".png";
             String path = GraphsStatsAgentSimEventsListener.CONTROLLER_IO.getIterationFilename(iteration, filename);
             int index = path.lastIndexOf("/");
             File outDir = new File(path.substring(0, index) + "/tripHistogram");
-            if (!outDir.exists()) outDir.mkdirs();
+            if (!outDir.exists()) Files.createDirectories(outDir.toPath());
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java]: Replace `outDir.exists` with `outDir.isDirectory`

**Comment by zishanbilal** [comment]
@carloscaldas  please review the changes.

**Pull request merged**